models/event.js

```diff
@@ -0,0 +1,46 @@
+let Event = null;
+const mongoose = require('mongoose');
+const connString = 'mongodb://localhost:27017/events';
+mongoose.connect(connString);
+
+mongoose.connection.on('connected',  () => {
+  console.log('Mongoose default connection open to ' + connString);
+});
+
+// If the connection throws an error
+mongoose.connection.on('error', (err) => {
+  console.log('Mongoose default connection error: ' + err);
+});
+
+// When the connection is disconnected
+mongoose.connection.on('disconnected', () => {
+  console.log('Mongoose default connection disconnected');
+});
+
+const Schema = mongoose.Schema;
+const eventSchema = new Schema(
+  {
+    _id: { type: Number, required: true },
+    title: { type: String, required: true, index: true},
+    description: { type: String, required: true },
+    date: { type: Date }
+  },
+  {
+    versionKey: false
+  }
+);
+
+Event = mongoose.model('Event', eventSchema);
+
+Event.ensureIndexes((err) => {
```

Comment:
We added this line because we found that without it the index was not created in the collection after our application execution.  We checked the index creation in the mongo shell with the command `db.collection.getIndexes()`.  After adding the .ensureIndexes() call we could see in mongo shell the index created.  I'm not sure if this is needed, or if  Mongoose ensures the use of the index even if it is not persisted